AppexClasses/8/OpportunityUpdate.cls

```diff
@@ -0,0 +1,24 @@
+public class OpportunityUpdate extends fflib_SObjectDomain 
+{
+   public OpportunityUpdate(List<Opportunity> sObjectList)
+   {
+       super(sObjectList);
+   }
+   
+   public class Constructor implements fflib_SObjectDomain.IConstructable 
+   {
+       public fflib_SObjectDomain construct(List<sObject> sObjectList) 
+       {
+           return new OpportunityUpdate(sObjectList);
+       }
+   }
+   
+   public override void onAfterInsert() 
+   {
+       for(Opportunity opp : (List<Opportunity>) Records) 
+       {                  
+           if(opp.BillToContact__c != null && opp.Manager__c == null)
+                   opp.Manager__c = opp.Account.Id;
```

Comment:
This statement will throw null pointer exception if account is not associated with Opportunity.